compiler-core/src/language_server/edits.rs

```diff
@@ -14,20 +14,33 @@ pub fn position_of_first_definition_if_import(
     module: &Module,
     line_numbers: &LineNumbers,
 ) -> Option<Position> {
-    // As "self.module.ast.definitions"  could be sorted, let's find the actual first definition by position.
-    let first_definition = module
-        .ast
-        .definitions
+    let TypedDefinitions {
+        imports,
+        constants,
+        custom_types,
+        type_aliases,
+        functions,
+    } = &module.ast.definitions;
+
+    // We first find the firts import by position
+    let first_import = imports.iter().min_by_key(|import| import.location)?;
+
+    // Then we need to make sure it actually comes before any other definition.
+    let import_is_first_definition = constants
         .iter()
-        .min_by(|a, b| a.location().start.cmp(&b.location().start));
-    let import = first_definition.and_then(get_import);
-    import.map(|import| src_span_to_lsp_range(import.location, line_numbers).start)
-}
+        .map(|constant| constant.location)
+        .chain(custom_types.iter().map(|custom_type| custom_type.location))
+        .chain(type_aliases.iter().map(|type_alias| type_alias.location))
+        .chain(functions.iter().map(|function| function.location))
+        .filter(|location| location.lt(&first_import.location))
```

Comment:
How come `.lt`?

Reply:
Too Gleam pilled and didn't think I could use operators with custom types